src/TLuaInterpreter.cpp

```diff
@@ -8146,7 +8138,27 @@ int TLuaInterpreter::tempAnsiColorTrigger(lua_State* L)
         }
     }
 
-    int triggerID = pLuaInterpreter->startTempColorTrigger(ansiFgColor, ansiBgColor, code);
+    if (lua_isstring(L, 3)) {
+        code = QString::fromUtf8(lua_tostring(L, 3));
+    } else if (lua_isfunction(L, 3)) {
+        // leave code as a null QString()
+    } else {
+        lua_pushfstring(L, "bad argument #3 type (code to run as a string or a function expected, got %s!)", luaL_typename(L, 3));
```

Comment:
Since this is an abort we need the function name (`tempAnsiColorTrigger:` prefix here).